src/interchange/src/avro.rs

```diff
@@ -1191,76 +1208,121 @@ impl DebeziumDeduplicationState {
             old_max_row: &'a usize,
             old_offset: &'a Option<i64>,
         }
-        let should_skip = match self.binlog_offsets.get_mut(file) {
-            Some((old_max_pos, old_max_row, old_offset)) => {
-                if (*old_max_pos, *old_max_row) >= (pos, row) {
-                    Some(SkipInfo {
-                        old_max_pos,
-                        old_max_row,
-                        old_offset,
-                    })
-                } else {
-                    // update the debezium high water mark
-                    *old_max_pos = pos;
-                    *old_max_row = row;
-                    *old_offset = coord;
+        let should_skip = if is_snapshot {
+            None
+        } else {
+            match self.binlog_offsets.get_mut(file) {
+                Some((old_max_pos, old_max_row, old_offset)) => {
+                    if (*old_max_pos, *old_max_row) >= (pos, row) {
+                        Some(SkipInfo {
+                            old_max_pos,
+                            old_max_row,
+                            old_offset,
+                        })
+                    } else {
+                        // update the debezium high water mark
+                        *old_max_pos = pos;
+                        *old_max_row = row;
+                        *old_offset = coord;
+                        None
+                    }
+                }
+                None => {
+                    // The extra lookup is fine - this is the cold path.
+                    self.binlog_offsets
+                        .insert(file.to_owned(), (pos, row, coord));
                     None
                 }
             }
-            None => {
-                // The extra lookup is fine - this is the cold path.
-                self.binlog_offsets
-                    .insert(file.to_owned(), (pos, row, coord));
-                None
-            }
         };
 
         match &mut self.full {
-            TrackFull::No => should_skip.is_none(),
-            TrackFull::SeenOffsets(seen_offsets) => {
-                if let Some(seen_offsets) = seen_offsets.get_mut(file) {
-                    let is_new = seen_offsets.insert((pos, row));
-
-                    match (is_new, should_skip) {
-                        // new item that correctly is past the highest item we've ever seen
-                        (true, None) => {}
-                        // new item that violates Debezium "guarantee" that the no new
-                        // records will ever be sent with a position below the highest
-                        // position ever seen
-                        (true, Some(skipinfo)) => {
-                            // TODO: This would be nicer if it included the wall-clock kafka event time
-                            warn!(
+            None => is_snapshot || should_skip.is_none(),
```

Comment:
this is saying that "current record is from a snapshot" is higher priority than "seems to be a duplicate"? If that's correct could you add that as an explicit comment here?